include/fastdds/dds/xtypes/dynamic_types/DynamicType.hpp

```diff
@@ -0,0 +1,166 @@
+// Copyright 2023 Proyectos y Sistemas de Mantenimiento SL (eProsima).
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#ifndef FASTDDS_DDS_XTYPES_DYNAMIC_TYPES_DYNAMIC_TYPE_HPP
+#define FASTDDS_DDS_XTYPES_DYNAMIC_TYPES_DYNAMIC_TYPE_HPP
+
+#include <memory>
+
+#include <fastdds/dds/core/ReturnCode.hpp>
+#include <fastdds/dds/xtypes/dynamic_types/AnnotationDescriptor.hpp>
+#include <fastdds/dds/xtypes/dynamic_types/DynamicTypeMember.hpp>
+#include <fastdds/dds/xtypes/dynamic_types/TypeDescriptor.hpp>
+#include <fastdds/dds/xtypes/dynamic_types/Types.hpp>
+#include <fastdds/dds/xtypes/dynamic_types/VerbatimTextDescriptor.hpp>
+#include <fastrtps/fastrtps_dll.h>
+
+namespace eprosima {
+namespace fastdds {
+namespace dds {
+
+class RTPS_DllAPI DynamicType : public std::enable_shared_from_this<DynamicType>
+{
+public:
+
+    using _ref_type = typename traits<DynamicType>::ref_type;
+
+    /*!
+     * Provides a summary of the state of this type overwriting a provided object (see [standard] 7.5.2.8.7)
+     * @param[inout] descriptor @ref TypeDescriptor reference where copied the information.
+     * @return standard @ref ReturnCode_t
```

Comment:
I am missing the possible return codes. The specification states that passing a nil argument returns RETCODE_BAD_PARAMETER. Being a getter, this is kind of confusing for me. If the specification is correct, then the `get_annotation` could also fail similarly but in that case is not stated in the spec.